lib/ancestry/materialized_path.rb

```diff
@@ -50,6 +50,14 @@ def sibling_conditions(object)
       t[ancestry_column].eq(node[ancestry_column])
     end
 
+    # idintifies leaves that belongs to the object (excluding itself)
+    def leaf_conditions(object)
```

Comment:
Does this method look useful to you, or are you simply following the existing pattern?
If you are following a pattern, could we do this method without the `object` parameter?

aside: I'd like to remove all the `*_conditions` methods but they are currently in the public API.

Reply:
Yes, I simply tried to respect the existing pattern as much as possible.

Just a question: I was assuming `object` parameter needs to be there to limit its scope within the subtree (just like `descendant_conditions` or `child_conditions` do), but would you rather prefer them to be an general methods that return all the nodes meeting with the conditions?

I'm totally fine to take that away, but just wanted to clarify.